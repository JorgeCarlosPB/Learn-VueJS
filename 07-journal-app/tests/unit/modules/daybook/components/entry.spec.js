

import { shallowMount } from "@vue/test-utils";
import Entry from '@/modules/daybook/components/MyEntry.vue'
import { journalState } from "../../../mock-data/test-journal-state";

describe ('' ,()=>{
    const mockRouter= {
        push: jest.fn()
    }
    //const wrapper = shallowMount(entry, {...props...global...})
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }

    })

    test('debe de hacer match con el snapshot', ()=>{
        
        expect (wrapper.html()).toMatchSnapshot();

    })

    test('debe de redireccionar al hacer click en el entry-contaienr', ()=>{

        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith({
                name:"entry",
                params: {
                    id:journalState.entries[0].id
                }
            
        })

    })

    test('pruebas en las propiedades computadas', ()=>{
        //wrapper.vm.<--------------ver las propiedades computadas
        //data: 23
        //month: Julio
        // yearDay: '2021, viernes'
        expect(wrapper.vm.day).toBe(23)
        expect(wrapper.vm.month).toBe('Junio')
        expect(wrapper.vm.yearDay).toBe('2022, Jueves')

    })
})