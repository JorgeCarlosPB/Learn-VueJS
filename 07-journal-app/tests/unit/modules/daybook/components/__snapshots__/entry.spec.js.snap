// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!--TÃ­tulo-->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">23</span><span class="mx-1 fs-5">Junio</span><span class="mx-2 fw-light">2022, Jueves</span></div>
  <div class="entry-description">Estos meses tengo mucho por aprender</div>
</div>
`;
